wdl-lint/tests/lints/matching-param-meta/source.wdl

```diff
@@ -54,3 +54,88 @@ workflow w {
 
     output {}
 }
+
+# Task with out-of-order parameter_meta
+task new_test {
+    meta {}
+
+    parameter_meta {
+        second: "This should be second"
+        first: "This should be first"
+    }
+
+    input {
+        # This should warn about incorrect ordering
+        String first
+        String second
+    }
+
+    command <<<>>>
+
+    output {}
+}
+
+struct Bar {
+    meta {}
+
+    parameter_meta {
+        param_b: "This should be after param_a"
+        param_a: "This should be first"
+    }
+
+    String param_a
+    String param_b
+}
+
+# This should trigger a InputSorting diagnostic,
```

Comment:
Why are we testing the `InputSorting` rule here?

Reply:
This was added at my request. I wanted to verify the interaction between these two rules.

IMO the desired behavior is that a parameter meta which _matches_ an "out of order" input section should not emit anything.

The next case (when neither are in the expected order) I am less sure about the desired behavior. I think anything we do there is potentially confusing.

Reply:
In that case, can we suppress the `InputSorting` diagnostic with a lint directive?

Reply:
I think it's better to show the `InputSorting` diagnostic in this test. Maybe we should add a comment at the top of the WDL file stating that this file is testing _both_ rules? Or more specifically, testing how they interact.

Am I missing something in terms of why you want this excepted?

Another idea is to move these cases to a dedicated test dir, `input-and-parameter-meta-sorting`?

Reply:
Yea this file is 141 lines long. I think the answer here is to break it up